config/rbac/metrics_reader_role.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: metrics-reader
```

Comment:
```
$ oc project openshift-vertical-pod-autoscaler
$ oc create serviceaccount test-sa
$ oc create -f - <<EOF
apiVersion: v1
kind: Secret
metadata:
  name: test-sa-token
  annotations:
    kubernetes.io/service-account.name: test-sa 
type: kubernetes.io/service-account-token
EOF
$ oc create clusterrolebinding test-sa --clusterrole=vpa-metrics-reader --serviceaccount=openshift-vertical-pod-autoscaler:test-sa
...
...
```
In your example, you used the `vpa-metrics-reader` cluster role. Should we rename this to that?

Reply:
Hmm, looks like we have a lot of old cruft in the bundle dir, which is where `vpa-metrics-reader` was coming from. Maybe we need to delete `bundle/manifests/` and re-generate it as part of the new test, to detect manifests which are out-of-date relative to the config.

And separate from that, I should have been more deliberate in following the upgrade steps instead of just wholesale copying what they put. Some of the names should be our own, but some of them, I'm not sure what we should do. The `metrics-reader` and `system:metrics-reader` cluster roles are not VPA specific. I'm not sure why `operator-sdk` is having us add them to our manifests (and probably doing the same to everyone else's manifests too). Maybe it's like a common dependency and any and all installed operators will provide them if they're not already there.

Reply:
It looks like `system:metrics-reader` comes from upstream VPA RBAC which we bring in https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/deploy/vpa-rbac.yaml#L1-L13

and `metrics-reader` comes from the final step of https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.38.0/

So in both cases, I think they are intended to be globally useful roles. The former for getting pod metrics (useful for HPA, VPA, etc.) and the latter for reaching metrics endpoints (for Prometheus, et al. to be able to scrape all metrics targets in the cluster with a single rolebinding).

I think we leave those two as-is.

Reply:
Oh, ok makes sense, but confused me on the naming. Thanks for getting to the bottom of that.

Reply:
Yeah, I'm glad you brought it up since it helped me realize about the stale bundle issue so we could get that cleaned up.